### PR TITLE
Resource partials

### PR DESCRIPTION
**Pull request by damacus**
# Description

- Require Chef 16 for resource partials
- Add resource partials for: MacOS, Linux, Java Home and Common as these are used in a multiple places

## Issues Resolved

List any existing issues this PR resolves

## Check List

- [ ] A summary of changes made is included in the CHANGELOG under `## Unreleased`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable.

### CODE REVIEW

**Review by Xorima — APPROVED** [review]

**Comment by ramereth** [comment]
Can we fix the failing suites before merging?

**Comment by damacus** [comment]
@ramereth that needs to be done in another pr and its not just a quick fix. Sadly I don't have time right now to figure out a nice way to pull down corretto

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,9 @@ This file is used to list changes made in each version of the Java cookbook.
 
 ## Unreleased
 
+- Require Chef 16 for resource partials
+- Add resoruce partials for: MacOS, Linux, Java Home and Common as these are used in a multiple places
```
Comment (ramereth) [inline CHANGELOG.md:8]:
```suggestion
- Add resource partials for: MacOS, Linux, Java Home and Common as these are used in a multiple places
```

File: resources/partial/_linux.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+        default: 1,
+        description: 'Alternatives priority to set for this Java'
+
+property :reset_alternatives, [true, false],
+        default: true,
+        description: 'Whether to reset alternatives before setting'
+
+property :default, [true, false],
+        default: true,
+        description: ' Whether to set this as the defalut Java'
```
Comment (ramereth) [inline resources/partial/_linux.rb:11]:
```suggestion
        description: ' Whether to set this as the default Java'
```

**Review by ramereth — DISMISSED** [review]
Typos

**Review by ramereth — DISMISSED** [review]

**Review by ramereth — APPROVED** [review]

**Comment by kitchen-porter** [comment]
Released as: [11.0.0](https://github.com/sous-chefs/java/releases/tag/11.0.0)

**Pull request merged**